src/libcrun/utils.c

```diff
@@ -1457,26 +1458,20 @@ set_home_env (uid_t id)
 
   stream = fopen ("/etc/passwd", "re");
   if (stream == NULL)
-    {
-      if (errno == ENOENT)
-        goto exit;
-
-      return -1;
-    }
+    goto error;
 
   for (;;)
     {
-      int ret;
       struct passwd *ret_pw = NULL;
 
       ret = fgetpwent_r (stream, &pwd, buf, buf_size, &ret_pw);
       if (UNLIKELY (ret != 0))
         {
           if (errno == ENOENT)
-            return 0;
+            goto error;
```

Comment:
this is not needed anymore, since the `errno != ERANGE` block below has the same body